File: src/2025/day03/src/test/java/ca/terrylockett/aoc2025/day03/TestDay03.java
package ca.terrylockett.aoc2025.day03;

import ca.terrylockett.aoccommon.resources.Resources;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TestDay03 {

	static final String TEST_FILE_NAME = "test-input.txt";

	static String testInput = "";

	@BeforeAll
	static void setup() {
		testInput = Resources.getInput(TEST_FILE_NAME).orElseThrow();
	}

	@Test
	void testDay03part1() {
		assertEquals(357, Day03Runner.part1(testInput));
	}
}

File: src/2023/day03/src/test/kotlin/ca/terrylockett/aoc2023/day03/TestDay03.kt
package ca.terrylockett.aoc2023.day03

import ca.terrylockett.aoccommon.inputfilefinder.InputFileFinder
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestDay03 {
    companion object {
        const val PART1_TEST_FILE_NAME = "test-input.txt"
    }

    @Test
    fun part1() {
        val inputFilePath = InputFileFinder.getInputFilePath(PART1_TEST_FILE_NAME).orElseThrow()
        val engine = EngineSchematic(inputFilePath)

        assertEquals(4361, engine.part1FindParts())
    }

    @Test
    fun part2() {
        val inputFilePath = InputFileFinder.getInputFilePath(PART1_TEST_FILE_NAME).orElseThrow()
        val engine = EngineSchematic(inputFilePath)

        assertEquals(467835, engine.part2FindGears())
    }
}
